ironforgedbot/commands/hiscore/constants.py

```diff
@@ -1,106 +1,751 @@
-    PRAYER = "Prayer"
-    MAGE = "Magic"
-    COOKING = "Cooking"
-    WC = "Woodcutting"
-    FLETCH = "Fletching"
-    FISH = "Fishing"
-    FM = "Firemaking"
-    CRAFT = "Crafting"
-    SMITH = "Smithing"
-    MINING = "Mining"
-    HERB = "Herblore"
-    AGI = "Agility"
-    THIEVING = "Thieving"
-    SLAYER = "Slayer"
-    FARM = "Farming"
-    RC = "Runecraft"
-    HUNTER = "Hunter"
-    CON = "Construction"
-
-    @classmethod
-    def has_value(cls, value):
-        return value in cls._value2member_map_
-
-
-class ACTIVITIES(StrEnum):
-    CLUES_BEGINNER = "Clue Scrolls (beginner)"
-    CLUES_EASY = "Clue Scrolls (easy)"
-    CLUES_MEDIUM = "Clue Scrolls (medium)"
-    CLUES_HARD = "Clue Scrolls (hard)"
-    CLUES_ELITE = "Clue Scrolls (elite)"
-    CLUES_MASTER = "Clue Scrolls (master)"
-    GOTR = "Rifts closed"
-    SIRE = "Abyssal Sire"
-    HYDRA = "Alchemical Hydra"
-    ARTIO = "Artio"
-    BARROWS = "Barrows Chests"
-    BRYO = "Bryophyta"
-    CALLISTO = "Callisto"
-    CALV = "Calvar'ion"
-    CERB = "Cerberus"
-    COX = "Chambers of Xeric"
-    COX_CM = "Chambers of Xeric: Challenge Mode"
-    CHAOS_ELEMENTAL = "Chaos Elemental"
-    CHAOS_FANATIC = "Chaos Fanatic"
-    ZILYANA = "Commander Zilyana"
-    CORP = "Corporeal Beast"
-    CRAZY_ARCHEOLOGIST = "Crazy Archaeologist"
-    PRIME = "Dagannoth Prime"
-    REX = "Dagannoth Rex"
-    SUPREME = "Dagannoth Supreme"
-    DERANGED_ARCHEOLOGIST = "Deranged Archaeologist"
-    DUKE = "Duke Sucellus"
-    GRAARDOR = "General Graardor"
-    MOLE = "Giant Mole"
-    GROTESQUE = "Grotesque Guardians"
-    HESPORI = "Hespori"
-    KQ = "Kalphite Queen"
-    KBD = "King Black Dragon"
-    KRAKEN = "Kraken"
-    KREE = "Kree'Arra"
-    KRIL = "K'ril Tsutsaroth"
-    LUNAR_CHEST = "Lunar Chests"
-    MIMIC = "Mimic"
-    NEX = "Nex"
-    NIGHTMARE = "Nightmare"
-    PROSANI_NIGHTMARE = "Phosani's Nightmare"
-    OBOR = "Obor"
-    MUSPAH = "Phantom Muspah"
-    SARACHNIS = "Sarachnis"
-    SCORPIA = "Scorpia"
-    SCURRIUS = "Scurrius"
-    SKOTIZO = "Skotizo"
-    SOL = "Sol Heredit"
-    SPINDEL = "Spindel"
-    TEMP = "Tempoross"
-    GAUNTLET = "The Gauntlet"
-    CG = "The Corrupted Gauntlet"
-    LEVIATHAN = "The Leviathan"
-    WHISPERER = "The Whisperer"
-    TOB = "Theatre of Blood"
-    TOB_HM = "Theatre of Blood: Hard Mode"
-    SMOKE_DEVIL = "Thermonuclear Smoke Devil"
-    TOA = "Tombs of Amascut"
-    TOA_EM = "Tombs of Amascut: Expert Mode"
-    ZUK = "TzKal-Zuk"
-    JAD = "TzTok-Jad"
-    VARDORVIS = "Vardorvis"
-    VENENATIS = "Venenatis"
-    VETION = "Vet'ion"
-    VORK = "Vorkath"
-    WT = "Wintertodt"
-    ZALCANO = "Zalcano"
-    ZULRAH = "Zulrah"
-
-    @classmethod
-    def has_value(cls, value):
-        return value in cls._value2member_map_
+from typing import Dict, NotRequired, TypedDict, Union
+
+EMPTY_SPACE = "⠀⠀"
+
+
+class Skill(TypedDict):
+    name: str
+    display_order: int
+    emoji_key: str
+    xp: int
+    level: int
+    points: int
```

Comment:
These are attributes of a player's data, not a Skill as a constant. Same for Activity